test/e2e/egress_test.go

```diff
@@ -256,80 +284,121 @@ func testEgressCRUD(t *testing.T, data *TestData) {
 					},
 				},
 			},
-			expectedEgressIP: "169.254.101.10",
-			expectedNodes:    sets.NewString(nodeName(0), nodeName(1)),
-			expectedTotal:    2,
+			expectedNodes: sets.NewString(nodeName(0), nodeName(1)),
+			expectedTotal: 2,
 		},
 		{
-			name:    "no matching Node",
-			ipRange: v1alpha2.IPRange{CIDR: "169.254.102.0/30"},
+			name: "no matching Node",
 			nodeSelector: metav1.LabelSelector{
 				MatchLabels: map[string]string{
 					"foo": "bar",
 				},
 			},
-			expectedEgressIP: "169.254.102.1",
-			expectedNodes:    sets.NewString(),
-			expectedTotal:    2,
+			expectedNodes: sets.NewString(),
+			expectedTotal: 2,
 		},
 	}
 	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			pool := data.createExternalIPPool(t, "pool-", tt.ipRange, tt.nodeSelector.MatchExpressions, tt.nodeSelector.MatchLabels)
-			defer data.crdClient.CrdV1alpha2().ExternalIPPools().Delete(context.TODO(), pool.Name, metav1.DeleteOptions{})
-
-			egress := data.createEgress(t, "egress-", nil, map[string]string{"foo": "bar"}, pool.Name, "")
-			defer data.crdClient.CrdV1alpha2().Egresses().Delete(context.TODO(), egress.Name, metav1.DeleteOptions{})
-			// Use Poll to wait the interval before the first run to detect the case that the IP is assigned to any Node
-			// when it's not supposed to.
-			err := wait.Poll(500*time.Millisecond, 3*time.Second, func() (done bool, err error) {
-				egress, err = data.crdClient.CrdV1alpha2().Egresses().Get(context.TODO(), egress.Name, metav1.GetOptions{})
-				if err != nil {
-					return false, err
-				}
-				if egress.Spec.EgressIP != tt.expectedEgressIP {
-					return false, nil
+		for _, ipR := range ipRanges {
+			ipVsersion := fmt.Sprintf("-IPv%d", ipR.ipVersion)
+			expectedTotal := ipR.expectedTotalIpNum
+			t.Run(tt.name+ipVsersion, func(t *testing.T) {
+				switch ipR.ipVersion {
+				case 4:
+					skipIfNotIPv4Cluster(t)
+				case 6:
+					// skipIfNotIPv6Cluster(t)
+					if clusterInfo.podV6NetworkCIDR == "" {
```

Comment:
isn't that the same check as `skipIfNotIPv6Cluster`?

Reply:
Yes, good catch, will fix it.

Reply:
Done